File: film-information-backstage-web/src/main/webapp/js/service/filmService.js
//服务层
app.service('filmService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../film/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../film/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../film/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../film/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../film/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../film/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../film/search.do?page='+page+"&rows="+rows, searchEntity);
	}

	this.countNumber=function(){
		return $http.get('../film/countNumber.do');
	}
});

File: film-information-service-interface/src/main/java/com/michael/service/FilmService.java
package com.michael.service;

import com.michael.group.FilmGroup;
import com.michael.pojo.Film;
import entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface FilmService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Film> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Film film);
	
	
	/**
	 * 修改
	 */
	public void update(Film film);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Film findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Film film, int pageNum, int pageSize);





	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(FilmGroup film, int pageNum, int pageSize);


	/**
	 *
	 * @return
	 */
	public Long countNumber();

}
